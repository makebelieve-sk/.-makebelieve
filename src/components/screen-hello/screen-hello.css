.welcome {
    display: flex;
    align-items: center;
    flex-direction: column;
    flex-wrap: wrap;
    width: 100%;
    height: 100%; 
}
.welcome_logo {
    margin: 0; 
    width: 200px;
    height: 100px;
    border: 1px solid #f0eed5;
    background: url("./img/melody-logo.png") center no-repeat;
}
.welcome_button {
    padding: 0;
    margin-top: 75px;
    margin-left: 20px;
    width: 110px;
    height: 150px;
    border: none;
    background: url("./img/button.png") center no-repeat;
}
    .welcome_button:hover {
        -webkit-transform: scale(1.05);
        transform: scale(1.05);
        cursor: pointer; 
    }
    .welcome_button:active {
        -webkit-transform: scale(1.04);
        transform: scale(1.04); 
    }
.ul-list {
    margin: 0;
    padding: 0;
}
.welcome_rules-title {
    margin: 0;
    font-family: "Fira Sans", "Arial", sans-serif;
    font-style: italic;
    font-weight: 300;
    font-size: 33px;
    color: #230d1a;
    margin-top: 50px;
}
.welcome_text {
    margin: 0;
    margin-top: 10px;
    text-align: center; 
}
.welcome_rules-list {
    margin: 0;
    padding: 0;
    list-style: none;
    text-align: center; 
}