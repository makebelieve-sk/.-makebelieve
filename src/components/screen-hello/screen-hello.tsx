import React from 'react';

import './screen-hello.css';

type ScreenHelloType = {
    time: number,
    mistakes: number,
    onStartButtonClick: () => any
};

export const ScreenHello: React.FC<ScreenHelloType> = ({ time, mistakes, onStartButtonClick }) => {
    return (
        <div className="welcome">
            <i className="welcome_logo"/>
            <button className="welcome_button" onClick={onStartButtonClick} />
            <h1 className="welcome_rules-title">Правила игры</h1>
            <p className="welcome_text">Правила просты:</p>
            <ul className="ul-list"> 
                <li className="welcome_rules-list">За { time } минут нужно ответить на все вопросы.</li>
                <li className="welcome_rules-list">Можно допустить { mistakes } ошибки.</li> 
            </ul>
            <p className="welcome_text">Удачи!</p>
        </div>
    );
};