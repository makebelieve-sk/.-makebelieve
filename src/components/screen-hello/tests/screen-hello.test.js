import React from 'react';
import renderer from 'react-test-renderer';
import Enzyme, { shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

import { ScreenHello } from '../screen-hello';

Enzyme.configure({ adapter: new Adapter() });

const setUp = (prop1, prop2, prop3) => shallow(
    <ScreenHello time={prop1} mistakes={prop2} onStartButtonClick={prop3} />
);

const mock = {
    time: 0,
    mistakes: 0,
    onStartButtonClick: jest.fn()
};

describe(`ScreenHello component`, () => {
    let component;

    beforeEach(() => {
        component = setUp(mock.time, mock.mistakes, mock.onStartButtonClick);
    });

    it(`should contain div wrapper`, () => {
        const wrapper = component.find('.welcome');
        expect(wrapper.length).toBe(1);
    });

    it(`ScreenHello correctly renders after relaunch`, () => {
        const tree = renderer
            .create(<ScreenHello
                time={mock.time}
                mistakes={mock.mistakes}
                onStartButtonClick={mock.onStartButtonClick}
            />)
            .toJSON();
    
        expect(tree).toMatchSnapshot();
    });
});