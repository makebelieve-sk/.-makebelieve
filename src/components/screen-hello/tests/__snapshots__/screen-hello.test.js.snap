// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ScreenHello component ScreenHello correctly renders after relaunch 1`] = `
<div
  className="welcome"
>
  <i
    className="welcome_logo"
  />
  <button
    className="welcome_button"
    onClick={[MockFunction]}
  />
  <h1
    className="welcome_rules-title"
  >
    Правила игры
  </h1>
  <p
    className="welcome_text"
  >
    Правила просты:
  </p>
  <ul
    className="ul-list"
  >
    <li
      className="welcome_rules-list"
    >
      За 
      0
       минут нужно ответить на все вопросы.
    </li>
    <li
      className="welcome_rules-list"
    >
      Можно допустить 
      0
       ошибки.
    </li>
  </ul>
  <p
    className="welcome_text"
  >
    Удачи!
  </p>
</div>
`;
